### PR TITLE
Fixed DHCP Policies issue on latest NDO

### PR DESCRIPTION
**Pull request by sajagana**
Note:

Added version check to handle the latest NDO dhcp polices issue.

### CODE REVIEW

**Review by anvitha-jain — DISMISSED** [review]

**Review by akinross — APPROVED** [review]
LGTM

File: mso/resource_mso_schema_template_bd.go [hunk 1]
```diff
@@ -141,10 +141,11 @@ func resourceMSOTemplateBD() *schema.Resource {
...
 			},
 			"dhcp_policy": &schema.Schema{
-				Type:        schema.TypeMap,
-				Optional:    true,
-				Description: "Configure dhcp policy in versions before NDO 3.2",
-				Computed:    true,
+				Type:          schema.TypeMap,
+				Optional:      true,
+				Description:   "Configure dhcp policy in versions before NDO 3.2",
+				Computed:      true,
```
Comment (shrsr) [inline mso/resource_mso_schema_template_bd.go:147]: I'm not sure of the behaviour of this resource for this attribute when it is computed for NDO < 3.2  and NDO >= 3.2 due to the version check. Hence this question:
Does this attribute get populated at all when a single dhcp_policy is created using the other attribute dhcp_policies?

**Review by shrsr — CHANGES_REQUESTED** [review]

File: mso/resource_mso_schema_template_bd.go (hunk 1 above)
Comment (sajagana) [reply mso/resource_mso_schema_template_bd.go:147]: dhcp_policies - Only for NDO >= 3.7 and dhcp_policy is only for MSO. So we should use proper attributes to configure the DHCP Policy based on the NDO/MSO version.

Also, the resource will set the dhcp_policy and dhcp_policies values based on the MSO/NDO version.

**Review by lhercot — APPROVED** [review]
LGTM

**Pull request merged**